integration-tests/bats/filter-branch.bats

```diff
@@ -427,3 +427,88 @@ SQL
     [[ "$output" =~ "| 3  | 3  | NULL   |" ]] || false
     [[ "$output" =~ "+----+----+--------+" ]] || false
 }
+
+@test "filter-branch: --all fails with working and staged changes" {
```

Comment:
I feel like you actually have to make a branch and have the working set on the other branch to test `--branches` and `--all`